lib/error-handler.js

```diff
@@ -95,9 +95,14 @@ function defaultErrorHandler (error, request, reply) {
 function fallbackErrorHandler (error, reply, cb) {
   const res = reply.raw
   const statusCode = reply.statusCode
+  const defaultContentType = 'application/json; charset=utf-8'
   let payload
   try {
-    const serializerFn = getSchemaSerializer(reply[kRouteContext], statusCode, reply[kReplyHeaders]['content-type'])
+    const serializerFn = getSchemaSerializer(
```

Comment:
It should return `false` in order to fix the issue.

Reply:
But this makes the output serializer be ignored. Check it out:

The changes I made to make it return false:
```sh
diff --git a/lib/schemas.js b/lib/schemas.js
index c9a603e8..45cfe0ba 100644
--- a/lib/schemas.js
+++ b/lib/schemas.js
@@ -165,7 +165,12 @@ function getSchemaSerializer (context, statusCode, contentType) {
 
       return false
     }
-    return responseSchemaDef[statusCode]
+
+    if (typeof responseSchemaDef[statusCode] === 'function') {
+      return responseSchemaDef[statusCode]
+    }
+
+    return false
   }
   const fallbackStatusCode = (statusCode + '')[0] + 'xx'
   if (responseSchemaDef[fallbackStatusCode]) {
@@ -178,7 +183,11 @@ function getSchemaSerializer (context, statusCode, contentType) {
       return false
     }
 
-    return responseSchemaDef[fallbackStatusCode]
+    if (typeof responseSchemaDef[fallbackStatusCode] === 'function') {
+      return responseSchemaDef[fallbackStatusCode]
+    }
+
+    return false
   }
   if (responseSchemaDef.default) {
     if (responseSchemaDef.default.constructor === Object && contentType) {
```

```sh
diff --git a/lib/error-handler.js b/lib/error-handler.js
index 558624fa..32cee751 100644
--- a/lib/error-handler.js
+++ b/lib/error-handler.js
@@ -95,14 +95,9 @@ function defaultErrorHandler (error, request, reply) {
 function fallbackErrorHandler (error, reply, cb) {
   const res = reply.raw
   const statusCode = reply.statusCode
-  const defaultContentType = 'application/json; charset=utf-8'
   let payload
   try {
-    const serializerFn = getSchemaSerializer(
-      reply[kRouteContext],
-      statusCode,
-      reply[kReplyHeaders]['content-type'] || defaultContentType
-    )
+    const serializerFn = getSchemaSerializer(reply[kRouteContext], statusCode, reply[kReplyHeaders]['content-type'])
     payload = (serializerFn === false)
       ? serializeError({
         error: statusCodes[statusCode + ''],
@@ -126,7 +121,7 @@ function fallbackErrorHandler (error, reply, cb) {
     payload = serializeError(new FST_ERR_REP_INVALID_PAYLOAD_TYPE(typeof payload))
   }
 
-  reply[kReplyHeaders]['content-type'] = defaultContentType
+  reply[kReplyHeaders]['content-type'] = 'application/json; charset=utf-8'
   reply[kReplyHeaders]['content-length'] = '' + Buffer.byteLength(payload)
 
   cb(reply, payload)
```

And the result of the tests:

```sh
npx tap test/serializeResponse.test.js
​ FAIL ​ test/serializeResponse.test.js
 ✖ should be equivalent

  test/serializeResponse.test.js                 
  126 |       t.error(err)                       
  127 |       t.equal(response.statusCode, 400)  
> 128 |       t.same(body, {                     
      | --------^                                
  129 |         statusCode: 400,                 
  130 |         error: 'Bad Request',            
  131 |         message: 'body must be object'   

  --- expected                         
  +++ actual                           
  @@ -1,5 +1,6 @@                      
   Object {                            
     "statusCode": 400,                
  +  "code": "FST_ERR_VALIDATION",     
     "error": "Bad Request",           
     "message": "body must be object", 
   }                                   

  test: test/serializeResponse.test.js serialize the response for a Bad Request
    error, as defined on the schema
  stack: |
    test/serializeResponse.test.js:128:9
    node_modules/simple-get/index.js:98:7
    IncomingMessage.<anonymous> (node_modules/simple-concat/index.js:8:13)

​ FAIL ​ test/serializeResponse.test.js 1 failed of 16 113.534ms
 ✖ should be equivalent



                         
  🌈 SUMMARY RESULTS 🌈  
                         
​ FAIL ​ test/serializeResponse.test.js 1 failed of 16 113.534ms
 ✖ should be equivalent


Suites:   ​1 failed​, ​1 of 1 completed
Asserts:  ​​​1 failed​, ​15 passed​, ​of 16
​Time:​   ​401.206ms
```

As you can see on the schema defined on the test (`test/serializeResponse.test.js:9`), `code` should not be in the response.

Reply:
The problem should be solved one by one.
When NO `content-type` is exist, there shouldn't be any serializer return.

The second one should be why the `content-type` is `undefined`.
You should never assign it directly with `application/json`.